cmd/nginx-ingress/main.go

```diff
@@ -445,6 +451,16 @@ func getSocketClient(sockPath string) *http.Client {
 	}
 }
 
+// validateLockName validates the name of a resource
+// It returns an error if the resource name is invalid
+func validateLockName(lock string) error {
```

Comment:
Maybe `validateResourceName` is a better function name ? That way we can re-use the validation function for other resources.

Reply:
@Rulox Do you suggest leaving `validateResourceName` in main.go and move to validate.go if/when we want to reuse it?

Reply:
Do you mean `validation.go` @Dean-Coakley ? I think main.go is better for this method.
 `validation.go` belongs to `pkg/apis/configuration/validation` This is, validation for CRDs (VS/VSR). 

Maybe we can also create another `.go` file for the validation of things that are not related with VS/VSR (or use `utils.go`). 

PS: there are more validation functions here in main.go so I don't think is a terrible idea to keep it here.